server/interaction/npc/npc.go

```diff
@@ -0,0 +1,100 @@
+package npc
+
+import (
+	"errors"
+
+	"github.com/joshprzybyszewski/cribbage/logic/scorer"
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/server/interaction"
+	"github.com/joshprzybyszewski/cribbage/utils/rand"
+)
+
+var npcs = map[model.PlayerID]npcLogic{
+	`dumbNPC`:       &dumbNPCLogic{},
+	`simpleNPC`:     &simpleNPCLogic{},
+	`calculatedNPC`: &calcNPCLogic{},
+}
+
+var _ interaction.Player = (*npcPlayer)(nil)
+
+type npcPlayer struct {
+	logic                npcLogic
+	id                   model.PlayerID
+	handleActionCallback func(a model.PlayerAction) error
+}
+
+// NewNPCPlayer creates a new NPC with specified type
+func NewNPCPlayer(pID model.PlayerID, cb func(a model.PlayerAction) error) (interaction.Player, error) {
+	l, ok := npcs[pID]
+	if !ok {
+		return &npcPlayer{}, errors.New(`not a valid npc mode`)
+	}
+	return &npcPlayer{
+		logic:                l,
+		id:                   pID,
+		handleActionCallback: cb,
+	}, nil
+}
+
+func (npc *npcPlayer) ID() model.PlayerID {
+	return npc.id
+}
+
+func (npc *npcPlayer) NotifyBlocking(b model.Blocker, g model.Game, s string) error {
+	a := npc.buildAction(b, g)
+	return npc.handleActionCallback(a)
+}
+
+// The NPC doesn't care about messages or score updates
+func (npc *npcPlayer) NotifyMessage(g model.Game, s string) error {
+	return nil
+}
+func (npc *npcPlayer) NotifyScoreUpdate(g model.Game, msgs ...string) error {
+	return nil
+}
+
+func (npc *npcPlayer) buildAction(b model.Blocker, g model.Game) model.PlayerAction {
+	a := model.PlayerAction{
+		GameID:    g.ID,
+		ID:        npc.ID(),
+		Overcomes: b,
+	}
+	switch b {
+	case model.DealCards:
+		a.Action = model.DealAction{
+			NumShuffles: rand.Intn(10) + 1,
+		}
+	case model.CribCard:
+		a.Action = npc.handleBuildCrib(g)
```

Comment:
🌵 can we rename this method? It would make more sense to me to read `getBuildCribAction` here

and similar below: `getPegAction`

Reply:
agreed - I will likely move these as well to the `logic` member on each NPC